### ISSUE TITLE
[BUG] TabularPredictor fastai cannot reproduce the same results when the training data and code are identical.

### ISSUE THREAD

**Issue by quinnnnnnnnn** [awslabs/autogluon#3376]
**Describe the bug**
AutoGluon Version:  0.8.1b20230619
Python Version:     3.9.16
Operating System:   Linux
Platform Machine:   x86_64
Platform Version:   #83-Ubuntu SMP Thu Jun 15 19:16:32 UTC 2023
Disk Space Avail:   6072.83 GB / 7999.42 GB (75.9%)
Train Data Rows:    1975249

Fitting model: NeuralNetFastAI ...
	-0.8927	 = Validation score   (-root_mean_squared_error)
	2248.13s	 = Training   runtime
	5.77s	 = Validation runtime

AutoGluon Version:  0.8.1b20230619
Python Version:     3.9.16
Operating System:   Linux
Platform Machine:   x86_64
Platform Version:   #83-Ubuntu SMP Thu Jun 15 19:16:32 UTC 2023
Disk Space Avail:   5439.37 GB / 7999.42 GB (68.0%)
Train Data Rows:    1975249

Fitting model: NeuralNetFastAI ...
	-0.8967	 = Validation score   (-root_mean_squared_error)
	2223.95s	 = Training   runtime
	5.4s	 = Validation runtime

**Expected behavior**
I come from a non-English speaking country, so there might be some improper expressions in my wording XD.

#1454 
I've reviewed the previous discussion, and the response indicated that under general circumstances, TabularPredictor would set the random seed to 0. However, the final results produced by fastai are always very unstable.This was also the case in version 0.7.I am looking forward to this feature being launched soon.
#2734 

**To Reproduce**
My dataset is quite large, so only the training code is provided below.

 hyperparameters = {"FASTAI": {}}
train = TabularDataset(train_df)

weights = np.linspace(0.8, 1.6, len(train))
sample_weight = "weight"
train[sample_weight] = weights

predictor = TabularPredictor(label=label, sample_weight=sample_weight).fit(
      train_data=train[pos_features_list + [sample_weight, label]],
      # tuning_data=tuning,
      num_gpus=1,
      hyperparameters=hyperparameters,
      # presets="best_quality",
      # num_bag_folds=10
  )

**Screenshots / Logs**
![myplot](https://github.com/autogluon/autogluon/assets/29170018/cd8cc43b-684f-45e7-9f47-3dcc445125e5)
![myplot2](https://github.com/autogluon/autogluon/assets/29170018/d63bf8d6-e9ff-46b0-a512-95428b1b6807)

**Installed Versions**

```python
INSTALLED VERSIONS
------------------
date                   : 2023-07-05
time                   : 14:18:08.464277
python                 : 3.9.16.final.0
OS                     : Linux
OS-release             : 5.15.0-76-generic
Version                : #83-Ubuntu SMP Thu Jun 15 19:16:32 UTC 2023
machine                : x86_64
processor              : x86_64
num_cores              : 32
cpu_ram_mb             : 225446
cuda version           : 11.515.105.01
num_gpus               : 1
gpu_ram_mb             : [12052]
avail_disk_size_mb     : 5062451
accelerate             : 0.16.0
autogluon              : 0.8.1b20230619
autogluon.common       : 0.8.1b20230619
autogluon.core         : 0.8.1b20230619
autogluon.eda          : 0.8.1b20230619
autogluon.features     : 0.8.1b20230619
autogluon.multimodal   : 0.8.1b20230619
autogluon.tabular      : 0.8.1b20230619
autogluon.timeseries   : 0.8.1b20230619
boto3                  : 1.26.93
catboost               : 1.2
defusedxml             : 0.7.1
evaluate               : 0.2.2
fairscale              : 0.4.13
fastai                 : 2.7.11
gluonts                : 0.13.2
hyperopt               : 0.2.7
imodels                : 1.3.17
ipython                : 8.12.2
ipywidgets             : 8.0.6
jinja2                 : 3.1.2
joblib                 : 1.2.0
jsonschema             : 4.17.3
lightgbm               : 3.3.5
matplotlib             : 3.6.3
missingno              : 0.5.2
mlforecast             : 0.7.3
networkx               : 3.1
nlpaug                 : 1.1.11
nltk                   : 3.8.1
nptyping               : 2.4.1
numpy                  : 1.23.5
omegaconf              : 2.2.3
onnxruntime-gpu        : 1.13.1
openmim                : 0.3.7
pandas                 : 1.5.3
phik                   : 0.12.3
Pillow                 : 9.4.0
psutil                 : 5.9.4
PyMuPDF                : 1.21.1
pyod                   : 1.0.9
pytesseract            : 0.3.10
pytorch-lightning      : 1.9.4
pytorch-metric-learning: 1.7.3
ray                    : 2.3.1
requests               : 2.28.2
scikit-image           : 0.19.3
scikit-learn           : 1.1.1
scikit-learn-intelex   : None
scipy                  : 1.10.1
seaborn                : 0.12.2
sentencepiece          : 0.1.97
seqeval                : 1.2.2
setuptools             : 67.8.0
shap                   : 0.41.0
skl2onnx               : 1.13
statsforecast          : 1.4.0
statsmodels            : 0.13.5
suod                   : 0.0.8
tabpfn                 : 0.1.9
tensorboard            : 2.11.2
text-unidecode         : 1.3
timm                   : 0.9.2
torch                  : 1.13.1.post200
torchmetrics           : 0.11.4
torchvision            : 0.14.1a0+59d9189
tqdm                   : 4.65.0
transformers           : 4.26.1
ujson                  : 5.7.0
vowpalwabbit           : 9.4.0
xgboost                : 1.7.4
yellowbrick            : 1.5

```

</details>

**Comment by Innixma** [comment]
Hello @quinnnnnnnnn, thanks for creating an issue! Indeed FastAI should produce consistent results.

Can you try to reproduce the inconsistency with only a small amount of data using a public dataset? This will help ensure that the problem isn't something subtle on your end.

It may also be good to try removing the extra logic in your code such as the sample weights to see if that is what is causing it.

**Comment by Innixma** [comment]
The Fastai model is now deterministic.
